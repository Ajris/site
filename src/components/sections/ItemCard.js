import React from "react";
import Card from "react-bootstrap/Card";

export default class ItemCard extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            title: props.title,
            text: props.text,
            footer: props.footer
        };
    }

    render() {
        return (
            <Card>
                <Card.Body>
                    <Card.Title>{this.state.title}</Card.Title>
                    <Card.Text>{this.state.text}</Card.Text>
                </Card.Body>
                <Card.Footer>
                    <small className="text-muted">{this.state.footer}</small>
                </Card.Footer>
            </Card>
        );
    }
}