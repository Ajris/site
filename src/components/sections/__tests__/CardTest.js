import React from "react";
import { create } from "react-test-renderer";
import ItemCard from "../ItemCard";

describe("ItemCard component", () => {
    test("it matches the snapshot", () => {
        const component = create(<ItemCard/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});