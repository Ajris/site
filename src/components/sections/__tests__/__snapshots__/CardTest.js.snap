// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard component it matches the snapshot 1`] = `
<div
  className="card"
>
  <h5
    className="card-header"
  />
  <img
    className="card-img-top"
  />
  <div
    className="card-body"
  >
    <div
      className="clearfix hidden-lg-up"
    >
      <p
        className="card-text"
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root button-card MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    />
  </button>
</div>
`;
